### PR TITLE
feat(kibbeh): NativeRadio component

### PR DESCRIPTION
**Pull request by Mvrs**
This PR:

- Adds NativeRadio Component
- Related to #2174

### CODE REVIEW

**Comment by BronzW** [comment]
Fix conflicts

**Comment by ajmnz** [comment]
Is tailwind forms really needed? I think this can be achieved without it and without svg too.

Let me know if you need a hand, I can take it from here if necessary

**Comment by Mvrs** [comment]
> Is tailwind forms really needed? I think this can be achieved without it and without svg too.
> 
> 
> 
> Let me know if you need a hand, I can take it from here if necessary

It's not necessary but I think it'll be more work without. Whats the best approach for a customized checkbook?

**Comment by ajmnz** [comment]
> It's not necessary but I think it'll be more work without. Whats the best approach for a customized checkbook?

Let’s do it without tailwind forms. Let me know if you want me to take over

**Comment by Mvrs** [comment]
> > It's not necessary but I think it'll be more work without. Whats the best approach for a customized checkbook?
> 
> Let’s do it without tailwind forms. Let me know if you want me to take over

I'm going to check tailwind components right quick. I know how to do it in pure CSS, but the lack of support for pseudo classes was why I went with forms. If I can't find anything I'll let you take over.

**Comment by Mvrs** [comment]
@ajmnz Cover 😎. Can't wait to see your solution.

**Comment by ajmnz** [comment]
@Mvrs Done! Revise it and we'll go ahead and merge

**Comment by Mvrs** [comment]
> @Mvrs Done! Revise it and we'll go ahead and merge

Coo. I'll get it started

File: kibbeh/src/ui/NativeRadio.tsx [hunk 1]
```diff
@@ -1,116 +1,90 @@
...
   title: string;
-  description: string;
-  id?: Settings;
+  subtitle: string;
   checked?: boolean;
-  radios?: Array<RadioOptions>;
-  children?: ReactNode;
+  onClick?: (id: number | undefined) => void;
+  num?: number;
 }
```
Comment (Mvrs) [inline kibbeh/src/ui/NativeRadio.tsx]: way better naming convention.

File: kibbeh/src/ui/NativeRadio.tsx [hunk 2]
```diff
@@ -1,116 +1,90 @@
...
       </div>
-      {children}
-    </React.Fragment>
+    </button>
   );
+};
+
+export interface NativeRadioControllerProps {
+  radios: NativeRadioProps[];
 }
```
Comment (Mvrs) [inline kibbeh/src/ui/NativeRadio.tsx]: such a proper way to handle your args. Nice way of cleaning up the logic.

File: kibbeh/src/ui/NativeRadio.tsx [hunk 3]
```diff
@@ -1,116 +1,90 @@
...
+          <div
+            className={`${
+              checked ? "bg-accent" : ""
+            } w-2 h-2 absolute top-2/4 left-2/4 rounded-full transform -translate-y-1/2 -translate-x-1/2 transition duration-100`}
+          ></div>
+          <div
+            className={`${
+              checked ? "border-accent" : "border-primary-300"
+            } border w-4 h-4 absolute top-2/4 left-2/4 rounded-full transform -translate-y-1/2 -translate-x-1/2 transition duration-100`}
+          ></div>
```
Comment (Mvrs) [inline kibbeh/src/ui/NativeRadio.tsx:53]: transform property to create the radio buttons? This is definitely better than using forms 😆

**Review by Mvrs — COMMENTED** [review]
In general, everything looked good. Very clean approach.

File: kibbeh/src/ui/NativeRadio.tsx [hunk 4]
```diff
@@ -1,116 +1,90 @@
...
-          <span
-            className={
-              radioSelected !== checked
-                ? `text-primary-100 font-bold`
-                : `${optionalChangedColor} font-bold`
-            }
+    <button
+      className="w-full flex px-3 py-2 bg-primary-900 rounded-8 justify-between group"
+      onClick={onClick ? () => onClick(num) : undefined}
+    >
```
Comment (Mvrs) [inline kibbeh/src/ui/NativeRadio.tsx:24]: On each card, there was a noticeable white outline that's not visibly pleasing. Not sure if you want to remove it or keep it. `focus:outline-none` would help

Comment (ajmnz) [reply kibbeh/src/ui/NativeRadio.tsx:24]: Yeah, Ben wants to keep that

**Pull request merged**